### ISSUE TITLE
feature : support defined environment variable in config.yaml

### ISSUE THREAD

**Issue by moorefu** [apache/apisix#1922]
i think we can support defined environment variable in config.yaml like this
```
nginx_config: 
    ...
    environment:
      TNS_ADMIN:
      DATACENTER_ID: 1
```

and render the template to genarate nginx.conf like this 
```
...
env APISIX_PROFILE;
env TNS_ADMIN;
env DATACENTER_ID=1;
...
```
it will be a useful feature.

**Comment by moonming** [comment]
sounds good to me, welcome PR

**Comment by spacewander** [comment]
Solved by #2174
